### PR TITLE
Adding Flag/Flags impl and test tags

### PR DESCRIPTION
**Pull request by bdlafleur**
This PR adds tests and implementation tags for the Flag and Flags classes.
---

## Checklist

- [x] This PR has only [one purpose or idea](https://terrapower.github.io/armi/developer/tooling.html#one-idea-one-pr).
- [x] [Tests](https://terrapower.github.io/armi/developer/tooling.html#test-it) have been added/updated to verify that the new/changed code works.

- [x] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [x] The commit message(s) follow [good practices](https://terrapower.github.io/armi/developer/tooling.html).

- [x] The [release notes](https://terrapower.github.io/armi/release/index.html) (location `doc/release/0.X.rst`) are up-to-date with any important changes.
- [x] The [documentation](https://terrapower.github.io/armi/developer/tooling.html#document-it) is still up-to-date in the `doc` folder.
- [x] No [requirements](https://terrapower.github.io/armi/developer/tooling.html#watch-for-requirements) were altered.
- [x] The dependencies are still up-to-date in `pyproject.toml`.

### CODE REVIEW

**Comment by bdlafleur** [comment]
There is no new source code added that would reduce coverage. Attributing failed checks to source spray.

**Comment by john-science** [comment]
Ignore the code coverage diff.

The service we use (coveralls.io) is an idiot. Someone has a PR out that increases code coverage from `main`, and coveralls can't tell the difference between `main` and a PR branch.

File: armi/reactor/flags.py [hunk 1]
```diff
@@ -181,6 +181,10 @@ def _toString(cls, typeSpec):
     """
     Make flag from string and fail if any unknown words are encountered.
 
+    .. impl:: Convert a flag to string
+        :id: I_ARMI_FLAG_TO_STR1
+        :implements: R_ARMI_FLAG_TO_STR
```
Comment (john-science) [inline armi/reactor/flags.py:186]: We should delete this.  You have this req on the `def toString()` method below, that's the correct one.

The REASON we can't have this here is that we can't put requirement stubs on private methods.  (Any method/class/function that starts with a `_` is "private" and won't appear in our API docs.)

Sorry, does that make sense?

File: armi/reactor/tests/test_flags.py [hunk 2]
```diff
@@ -27,8 +27,16 @@ def test_fromString(self):
         self.assertEqual(flags.Flags.fromStringIgnoreErrors("invalid"), flags.Flags(0))
 
     def test_toString(self):
+        """
+        Convert flag to and from string for serialization.
```
Comment (john-science) [inline armi/reactor/tests/test_flags.py:31]: I like what you're doing here.

Maybe we should change the NAME of this test so it says "to" and "from"?

`test_flagsToAndFromString()` or something like that?

File: armi/utils/tests/test_flags.py [hunk 3]
```diff
@@ -69,14 +69,46 @@ class F(Flag):
         f2 = F.from_bytes(array)
         self.assertEqual(f, f2)
 
-    def test_collision(self):
-        """Make sure that we catch value collisions."""
+    def test_collision_extension(self):
+        """Ensure the set of flags cannot be programmatically extended if duplicate created.
+
+        TODO: This test should fail when extend bug is fixed.
```
Comment (john-science) [inline armi/utils/tests/test_flags.py:75]: Okay, this is going to be odd, but it's in our docs.

I understand there are a bunch of `TODO`s in ARMI, I get that.

But our rule is that no NEW `TODO` statement can be added to the repo.  If you are trying to add a new TODO, that means either:

* You need to complete your code before your PR, it's not complete yet.
* There is an existing bug in the repo, in which case we need to open a ticket.

The _goal_ of the rule is to improve the code quality of the repo, which was problematic in the past.  Does that make sense?

Thanks!

Comment (bdlafleur) [reply armi/utils/tests/test_flags.py:75]: @john-science I've opened a new ticket with this issue noted ([#1469](https://github.com/terrapower/armi/issues/1469)).

**Review by john-science — APPROVED** [review]

**Pull request merged**